components/Users/AuthSettings.js

```diff
@@ -0,0 +1,135 @@
+
+import {
+  InlineSpinner,
+  Loader,
+  Radio
+} from '@project-r/styleguide'
+
+import {
+  InteractiveSection,
+  SectionTitle
+} from '../Display/utils'
+
+import withT from '../../lib/withT'
+
+import { SUPPORTED_TOKEN_TYPES } from '../constants'
+
+export const UPDATE_PREFERRED_FRIST_FACTOR = gql`
+  mutation updatePreferredFirstFactor(
+    $userId: ID!
+    $tokenType: SignInTokenType!
+  ) {
+    preferredFirstFactor(
+      userId: $userId
+      tokenType: $tokenType
+    ) {
+      id
+      enabledFirstFactors
+      preferredFirstFactor
+    }
+  }
+`
+
+export const GET_AUTH_SETTINGS = gql`
+  query getAuthSettings($userId: String) {
+    user(slug: $userId) {
+      id
+      enabledFirstFactors
+      preferredFirstFactor
+    }
+  }
+`
+
+class UpdatePreferredFirstFactor extends Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      value: this.props.user.preferredFirstFactor
+    }
+  }
+
+  render() {
+    const {
+      user: {
+        id: userId,
+        enabledFirstFactors,
+        preferredFirstFactor
+      },
+      t
+    } = this.props
+    
+    return (
+      <Mutation
+        mutation={UPDATE_PREFERRED_FRIST_FACTOR}
+        variables={{ userId, tokenType: this.state.value }}
+        refetchQueries={[{
+          query: GET_AUTH_SETTINGS,
+          variables: { userId }
+        }]}
+        awaitRefetchQueries={true}
+      >
+        {(mutation, { loading }) => {
+          return (
+            <InteractiveSection>
+              {loading && (
+                <span style={{ float: 'right' }}>
+                  <InlineSpinner size={32} />
+                </span>
+              )}
+              <SectionTitle>
+                Bevorzugte Anmeldeart
+              </SectionTitle>
+              <form>
+                {SUPPORTED_TOKEN_TYPES.map(tokenType => {
+                  const disabled = !enabledFirstFactors.includes(tokenType)
+                  
+                  return (
+                    <Fragment key={tokenType}>
+                      <Radio
+                        checked={tokenType === preferredFirstFactor}
+                        disabled={disabled || loading}
+                        onChange={() => {
+                          this.setState(
+                            () => ({ value: tokenType }),
+                            mutation
+                          )
+                        }}
+                      >
+                        {t(`account/authSettings/firstfactor/${tokenType}/label`)}
+                      </Radio>
+                      <br />
+                    </Fragment>
+                  )
+                })}
+              </form>
+            </InteractiveSection>
+          )
+        }}
+      </Mutation>
+    )
+  }
+}
+
+const AuthSettings = ({ t, userId }) => (
+  <Query query={GET_AUTH_SETTINGS} variables={{ userId }}>
+    {({ data, loading, error }) => (
+      <Loader
+        loading={loading && !data && !data.user}
```

Comment:
Setting loading to `true` for `!data && !data.user` could cause an errors to not display. The `loading` prop wins over `error` and an error could cause `data.user` to be falsy.

Just removing this check should do. No additional checks should be needed—`render` is only called once it successfully loaded. Let me know if I'm wrong.

```suggestion
        loading={loading}
```